packages/core/src/components/DatePicker/DatePicker.styled.tsx

```diff
@@ -1,7 +1,47 @@
-import { styled } from '@medly-components/utils';
-import { default as MedlyInput } from '../Input';
+import { SvgIcon } from '@medly-components/icons';
+import { css, styled, WithThemeProp } from '@medly-components/utils';
+import { OuterWrapper } from '../TextField/Styled';
+import { InnerWrapper } from '../TextField/Styled/InnerWrapper.styled';
+import { StyleProps } from './types';
 
-export const Input = styled(MedlyInput)`
-    user-select: none;
-    cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'inherit')};
+type State = 'default' | 'active' | 'error' | 'disabled';
+
+const getStyleForIcon = ({ theme, variant }: StyleProps & WithThemeProp, state: State) => {
+    const { datePicker } = theme;
+    const {
+        icon: { [state]: iconStyle }
+    } = datePicker[variant];
+    return css`
+        ${SvgIcon} {
+            border-radius: 50%;
+            padding: 0.8rem;
+            * {
+                fill: ${iconStyle.color};
+            }
+            &:hover {
+                background-color: ${iconStyle.backgroundColor};
+                * {
+                    fill: ${iconStyle.hoverColor};
```

Comment:
we have to add :active also to denote pressed state

Reply:
I guess that would get covered with this [state](https://github.com/medly/medly-components/blob/b303fe205db5b28a31a08694d37cc0ae22199b30/packages/core/src/components/DatePicker/DatePicker.styled.tsx#L34). wdyt?